src/html.cr

```diff
@@ -30,7 +30,9 @@ module HTML
   # HTML.escape("Crystal & You") # => "Crystal &amp; You"
   # ```
   def self.escape(string : String) : String
-    string.gsub(SUBSTITUTIONS)
+    io = IO::Memory.new
```

Comment:
`String.build` is the idiomatic way to build strings. If alternatives are faster, then `String.build` should be improved (global overall benefit), instead of using alternatives in stdlib (duplications).